pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionGroupConsumer.java

```diff
@@ -39,4 +58,19 @@
    */
   MessageBatch fetchMessages(StreamPartitionMsgOffset startOffset, StreamPartitionMsgOffset endOffset, int timeoutMs)
       throws TimeoutException;
+
+  /**
+   * Checkpoints the consumption state of the stream partition group in the source
+   *
+   * This is useful in systems that require preserving consumption state on the source in order to resume or replay
+   * consumption of data.
+   * The offset returned will be used for offset comparisons and persisted to the ZK segment metadata. Hence, the
```

Comment:
Are the offsets returned for local comparison (within the server while catching up, for example) different from offsets returned for comparisons with zk segment metadata?

Reply:
catching up? catching up happens only in non-winner replicas, iiuc. This method won't even be invoked as only winning replicas will checkpoint to source. 

offset returned will be persisted in ZK as the endOffset of this segment and used as startingOffset of the next segment. It should still be comparable within the server. This system-specific implementation of `StreamPartitionMsgOffset` should provide these guarantees.

Reply:
I have updated the doc based on our offline conversation . Please let me know if you need more changes here.